src/features/views/components/ViewColumnDialog/ColumnGallery.tsx

```diff
@@ -12,104 +21,242 @@ import choices, { ColumnChoice } from './choices';
   existingColumns,
   onAdd,
+  onClose,
   onConfigure,
 }) => {
   const intl = useIntl();
   const isMobile = useMediaQuery((theme: Theme) =>
     theme.breakpoints.down('sm')
   );
 
+  const [isSearching, setIsSearching] = useState(false);
+  const [searchString, setSearchString] = useState('');
+  const [searchResults, setSearchResults] = useState<
+    (ColumnChoice | undefined)[]
+  >([]);
+
+  //list of objects with localized strings to use in the fuse-search
+  const searchObjects = choices.map((choice) => ({
+    description: intl.formatMessage({
+      id: `misc.views.columnDialog.choices.${choice.key}.description`,
+    }),
+    key: choice.key,
+    keywords: intl.formatMessage({
+      id: `misc.views.columnDialog.choices.${choice.key}.keywords`,
+    }),
+    title: intl.formatMessage({
+      id: `misc.views.columnDialog.choices.${choice.key}.title`,
+    }),
+  }));
+
+  const search = () => {
+    const fuse = new Fuse(searchObjects, {
+      keys: ['description', 'keywords', 'title'],
+      threshold: 0.4,
+    });
+
+    //get the keys from the search results
+    const keys = fuse
+      .search(searchString)
+      .map((fuseResult) => fuseResult.item.key);
+
+    //get the choice objects with matching keys
+    const results = keys.map((key) =>
+      choices.find((choice) => choice.key === key)
+    );
+
+    setSearchResults(results);
+  };
+
   const choiceContainerRef = useRef<HTMLDivElement>();
 
   return (
-    <Box display="flex" height="100%">
-      <Box
-        alignItems="center"
-        display={isMobile ? 'none' : 'flex'}
-        flexDirection="column"
-        justifyContent="space-between"
-        sx={{
-          overflowY: 'scroll',
-        }}
-        width="20%"
-      >
-        <List>
-          {categories.map((category, index) => (
-            <ListItem
-              key={index}
-              onClick={() => {
-                if (choiceContainerRef.current) {
-                  const element = choiceContainerRef.current.querySelector(
-                    `#category-${index}`
-                  );
-                  element?.scrollIntoView({ behavior: 'smooth' });
-                }
-              }}
-              sx={{ cursor: 'pointer', paddingY: 2 }}
-            >
-              <Typography>
+    <Box display="flex" flexDirection="column" height="100%">
+      <Box display="flex" justifyContent="space-between" padding={2}>
+        <Typography variant="h4">
+          <Msg id="misc.views.columnDialog.gallery.columns" />
+        </Typography>
+        <Box alignItems="center" display="flex">
+          <TextField
+            InputProps={{
+              endAdornment: <Search color="secondary" />,
+            }}
+            onChange={(evt) => {
+              setSearchString(evt.target.value);
+              if (evt.target.value === '') {
+                setIsSearching(false);
+                return;
+              }
+              setIsSearching(true);
+              search();
```

Comment:
It should not be necessary to search here. Searching can happen on the next re-render, so that no extra state is required.

Reply:
i am not sure i understand, but i probably agree when i do :)